src/Core/functions.php

```diff
@@ -526,3 +531,88 @@ function toBool(mixed $value): ?bool
 
     return null;
 }
+
+/**
+ * Converts a value to a native DateTime object.
+ *
+ *  integer - value is treated as a Unix timestamp
+ *  string - value is treated as a ISO-8601 (Atom) formatted timestamp
+ *  Other values returns as null.
+ *
+ * @param mixed $value The value to convert to DateTime.
+ * @return \Cake\I18n\DateTime|null Returns a DateTime object if parsing is successful, or NULL otherwise.
+ * @since 5.1.0
+ */
+function toDateTime(mixed $value): ?DateTime
+{
+    if ($value instanceof DateTime) {
+        return $value;
+    }
+
+    if ($value instanceof DateTimeInterface) {
+        return DateTime::parse($value);
+    }
+
+    if (is_int($value)) {
```

Comment:
With the recent changes I made to Chronos::createFromTimestamp() it accepts floats too so this check could be updated to allow floats too. Also considering values from query strings, they would be numeric strings, so do we want to just use a `is_numeric()` check here?

Reply:
I've made the change to `is_numeric` but I need to update the tests now. I bumped the `cakephp/chronos` dependency to 3.1 to get @ADmad's change, just checking that was OK to do?